sprockets_end_to_end_test.py

```diff
@@ -0,0 +1,84 @@
+"""End-to-end tests for all of Sprockets."""
+
+import os
+import shutil
+import tempfile
+import unittest
+
+import test_driver
+
+# A default manifest to use in the tests below. If used, make sure
+# the STL file under test uses the filename "sprockets_end_to_end.stl" to match
+# what's in the manifest here.
+_DEFAULT_MANIFEST_FILENAME = 'sprockets_end_to_end.test'
+
+# External event functions that can be used in the tests below.
+_NOOP_PY_FILENAME = 'noop.py'
+
+
+class EndToEndTest(unittest.TestCase):
+
+  @classmethod
+  def setUpClass(cls):
+    # Need to capture this before the tests start running, otherwise the
+    # GetTestData method won't be able to find the end_to_end_test_data/
+    # folder.
+    cls.sprockets_dir = os.path.abspath(os.path.dirname(__file__))
+
+  def setUp(self):
+    self.tmpdir = tempfile.mkdtemp()
+    self.orig_cwd = os.getcwd()
+    os.chdir(self.tmpdir)
+
+    default_manifest_contents = self.GetTestData(_DEFAULT_MANIFEST_FILENAME)
+    self.CreateFile(_DEFAULT_MANIFEST_FILENAME, default_manifest_contents)
```

Comment:
Why do you need to create new files for test? Why can't you run with original files?

Reply:
Done